packages/medusa/src/utils/validator.ts

```diff
@@ -1,6 +1,34 @@
 import { ClassConstructor, plainToInstance } from "class-transformer"
 import { validate, ValidationError, ValidatorOptions } from "class-validator"
 import { MedusaError } from "medusa-core-utils"
+import { Constructor } from "@medusajs/types"
+
+const extendedValidators: Map<string, Constructor<any>> = new Map()
+
+/**
+ * When overriding a validator, you can register it to be used instead of the original one.
+ * For example, the place where you are overriding the core validator, you can call this function
+ * @example
+ * ```ts
+ * // /src/api/routes/admin/products/create-product.ts
+ * import { registerOverriddenValidators } from "@medusajs/medusa"
+ * import { AdminPostProductsReq as MedusaAdminPostProductsReq } from "@medusajs/medusa/dist/api/routes/admin/products/create-product"
+ * import { IsString } from "class-validator"
+ *
+ * class AdminPostProductsReq extends MedusaAdminPostProductsReq {
```

Comment:
I really like this idea. Think it will make for a smoother validator DX.

question(s):
- As far as I can see, you must register the "overriding validator" by passing its exact name from the core. I think this is OK. However, there are two minuscule friction points: you need to know what validator to pull from the core and be careful about naming your own validator. Again, I think this is fine, but I was just wondering if there's a way to eliminate these requirements? This is non-blocking, and I am mostly curious to hear your thoughts about it.
- Secondly, something that maybe goes along with my first comment, when you extend a core validator, are you not implicitly already stating, that you want this to override the core one? If so, would it be better to "hide" the overriding from the developer i.e. let it happen under the hood?

Reply:
I will approve the PR, as these are non-blocking and can be solved by having the approach well-documented :)

Reply:
Makes sense!

> the user must already know which one to extend and then extends it

Yup, this is also why I think this approach is reasonable. I was just curious to hear if you had thought of a different approach that was less coupled with our naming conventions.